src/Features/CSharp/Portable/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs

```diff
@@ -13,134 +13,150 @@
+        public const string CS0308 = "CS0308";
 
         /// <summary>
         /// 'A' is inaccessible due to its protection level
         /// </summary>
-        private const string CS0122 = "CS0122";
+        public const string CS0122 = "CS0122";
 
         /// <summary>
         /// The using alias 'A' cannot be used with type arguments
         /// </summary>
-        private const string CS0307 = "CS0307";
+        public const string CS0307 = "CS0307";
 
         /// <summary>
         /// 'A' is not an attribute class
         /// </summary>
-        private const string CS0616 = "CS0616";
+        public const string CS0616 = "CS0616";
 
         /// <summary>
         ///  No overload for method 'X' takes 'N' arguments
         /// </summary>
-        private const string CS1501 = "CS1501";
+        public const string CS1501 = "CS1501";
 
         /// <summary>
         /// cannot convert from 'int' to 'string'
         /// </summary>
-        private const string CS1503 = "CS1503";
+        public const string CS1503 = "CS1503";
 
         /// <summary>
         /// XML comment on 'construct' has syntactically incorrect cref attribute 'name'
         /// </summary>
-        private const string CS1574 = "CS1574";
+        public const string CS1574 = "CS1574";
 
         /// <summary>
         /// Invalid type for parameter 'parameter number' in XML comment cref attribute
         /// </summary>
-        private const string CS1580 = "CS1580";
+        public const string CS1580 = "CS1580";
 
         /// <summary>
         /// Invalid return type in XML comment cref attribute
         /// </summary>
-        private const string CS1581 = "CS1581";
+        public const string CS1581 = "CS1581";
 
         /// <summary>
         /// XML comment has syntactically incorrect cref attribute
         /// </summary>
-        private const string CS1584 = "CS1584";
+        public const string CS1584 = "CS1584";
 
         /// <summary>
         /// Type 'X' does not contain a valid extension method accepting 'Y'
         /// </summary>
-        private const string CS1929 = "CS1929";
+        public const string CS1929 = "CS1929";
 
         /// <summary>
         /// Cannot convert method group 'X' to non-delegate type 'Y'. Did you intend to invoke the method?
         /// </summary>
-        private const string CS0428 = "CS0428";
+        public const string CS0428 = "CS0428";
 
         /// <summary>
         ///  There is no argument given that corresponds to the required formal parameter 'X' of 'Y'
         /// </summary>
-        private const string CS7036 = "CS7036";
-
-        public override ImmutableArray<string> FixableDiagnosticIds
-        {
-            get
-            {
-                return ImmutableArray.Create(
-                    CS0103,
-                    CS0246,
-                    CS0305,
+        public const string CS7036 = "CS7036";
+
+        public static ImmutableArray<string> FixableTypeIds =
```

Comment:
Need explanatory comments on these. I don't know what the difference between a "FixableType" and a "FixableDiagnostic" is.